src/metabase/db/setup.clj

```diff
@@ -54,39 +54,37 @@
    data-source :- javax.sql.DataSource
    direction   :- s/Keyword]
   (with-open [conn (.getConnection data-source)]
-    ;; TODO: this try has no catch. We need to make sure the catches are at the right level and possibly remove this
```

Comment:
This diff is large but quite simple. There is an inner `try` that catches two errors, rolls back and then rethrows. An outer try had no `catch` so i just removed the outer try to match the current behavior of not handling errors. This is the `migrate!` function so throwing errors seems like the right thing to do (and is the current behavior)